src/Compilers/CSharp/Portable/Syntax/Syntax.xml

```diff
@@ -1720,12 +1720,52 @@
   <AbstractNode Name="PatternSyntax" Base="CSharpSyntaxNode" />
   <Node Name="DeclarationPatternSyntax" Base="PatternSyntax">
     <Kind Name="DeclarationPattern" />
-    <Field Name="Type" Type="TypeSyntax" />
+    <Field Name="Type" Type="TypeSyntax"/>
     <Field Name="Designation" Type="VariableDesignationSyntax">
       <Kind Name="SingleVariableDesignation"/>
       <Kind Name="DiscardDesignation"/>
     </Field>
   </Node>
+  <Node Name="DeconstructionPatternSyntax" Base="PatternSyntax">
+    <Kind Name="DeconstructionPattern" />
+    <Field Name="Type" Type="TypeSyntax" Optional="true" />
+    <Field Name="OpenParenToken" Type="SyntaxToken">
+      <Kind Name="OpenParenToken"/>
+    </Field>
+    <Field Name="SubPatterns" Type="SeparatedSyntaxList&lt;SubpatternElementSyntax&gt;"/>
+    <Field Name="CloseParenToken" Type="SyntaxToken">
+      <Kind Name="CloseParenToken"/>
+    </Field>
+    <Field Name="PropertySubpattern" Type="PropertySubpatternSyntax" Optional="true" />
+    <Field Name="Designation" Type="VariableDesignationSyntax" Optional="true">
+      <Kind Name="SingleVariableDesignation"/>
+      <Kind Name="DiscardDesignation"/>
+    </Field>
+  </Node>
+  <Node Name="SubpatternElementSyntax" Base="CSharpSyntaxNode">
```

Comment:
>SubpatternElementSyntax [](start = 14, length = 23)

It feels like this could be just a flavor of DeconstructionPattern node if the ```(sub-pattern list)``` was optional. In fact, both cold be merged with DeclarationPatternSyntax, but that would require renaming the node and we cannot do this now.

Reply:
I understand your suggestion. I am reluctant to declare a syntax node in which all parts are optional.